pkg/controller/operators/olm/operator.go

```diff
@@ -296,6 +301,50 @@ func NewOperator(logger *logrus.Logger, crClient versioned.Interface, opClient o
 	return op, nil
 }
 
+func (a *Operator) syncAPIService(obj interface{}) (syncError error) {
+	apiSvc, ok := obj.(*apiregistrationv1.APIService)
+	if !ok {
+		a.Log.Debugf("wrong type: %#v", obj)
+		return fmt.Errorf("casting APIService failed")
+	}
+
+	logger := a.Log.WithFields(logrus.Fields{
+		"id":     queueinformer.NewLoopID(),
+		"apiSvc": apiSvc.GetName(),
+	})
+	logger.Info("syncing APIService")
+
+	serviceLabels := apiSvc.GetLabels()
+	owningNamespace, owned := serviceLabels[ownerutil.OwnerNamespaceKey]
+	if owned {
+		_, err := a.lister.CoreV1().NamespaceLister().Get(owningNamespace)
+		if k8serrors.IsNotFound(err) {
+			logger.Debug("Deleting api service since owning namespace is not found")
+			syncError = a.OpClient.DeleteAPIService(apiSvc.GetName(), &metav1.DeleteOptions{})
+			return
+		}
+	}
+
+	owningCSVName, owned := serviceLabels[ownerutil.OwnerKey]
+	if owned {
+		_, err := a.lister.OperatorsV1alpha1().ClusterServiceVersionLister().ClusterServiceVersions(owningNamespace).Get(owningCSVName)
+		if k8serrors.IsNotFound(err) {
+			logger.Debug("Deleting api service since owning CSV is not found")
+			syncError = a.OpClient.DeleteAPIService(apiSvc.GetName(), &metav1.DeleteOptions{})
```

Comment:
We also generate some rolebindings in kube-system for APIServices that we should clean up

Reply:
I still need to do this. But I admit I'm less concerned about it since I've never seen any stray RBAC cause package server install issues.